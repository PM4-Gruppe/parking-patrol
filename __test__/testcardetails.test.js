import renderer from 'react-test-renderer';
import { useRouter } from 'next/router';
import { mocked } from 'jest-mock';
import { ApolloProvider } from '@apollo/client';
import apolloClient from '../src/lib/apollo';
import Cardetails from '../src/pages/cardetails';

jest.mock('next/router', () => ({
    useRouter: jest.fn(),
}));

console.error = jest.fn();

describe('Cardetails', () => {
    beforeEach(() => {
        mocked(useRouter).mockReturnValue({
            query: { carpath: 'test-carpath.jpg' },
            back: jest.fn(),
        });
    });

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('should match snapshot', () => {
        const tree = renderer
            .create(
                <ApolloProvider client={apolloClient}>
                    <Cardetails />
                </ApolloProvider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
