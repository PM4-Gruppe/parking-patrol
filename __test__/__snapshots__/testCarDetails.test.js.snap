// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cardetails should match snapshot 1`] = `
<div
  className=""
>
  <p
    className="m-5 text-white text-3xl"
  >
    Car Details
  </p>
  <div
    className="flex flex-col"
  >
    <div
      className="m-5 p-1 grow rounded-lg bg-neutral-700 overflow-auto drop-shadow-xl"
    >
      <img
        alt="Picture of the author"
        data-nimg="1"
        decoding="async"
        height={500}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fapi%2Fimage%2Ftest-carpath.jpg&w=1080&q=75"
        srcSet="/_next/image?url=%2Fapi%2Fimage%2Ftest-carpath.jpg&w=640&q=75 1x, /_next/image?url=%2Fapi%2Fimage%2Ftest-carpath.jpg&w=1080&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={500}
      />
    </div>
  </div>
  <div
    className="m-5"
  >
    <button
      className="p-2 bg-green-700 hover:bg-green-800 text-white rounded-lg my-2"
      onClick={[Function]}
      type="button"
    >
      Zur√ºck
    </button>
  </div>
</div>
`;
